azurelinuxagent/common/logger.py

```diff
@@ -260,24 +263,6 @@ def _do_rollover(self):
                 # Removing the plain log file as already archived.
                 os.remove(self.path)
             else:
```

Comment:
I think you forgot to call the function here in else :)

Reply:
Not required here. I optimized it above itself, based on `_should_archive_backup_files` attribute.